include/arch/arc/v2/irq.h

```diff
@@ -104,8 +106,8 @@ static inline void arch_isr_direct_footer(int maybe_swap)
 	    z_arc_v2_aux_reg_read(_ARC_V2_AUX_IRQ_HINT)) {
 		z_arc_v2_aux_reg_write(_ARC_V2_AUX_IRQ_HINT, 0);
 	}
-#ifdef CONFIG_TRACING
-	z_sys_trace_isr_exit();
+#ifdef CONFIG_TRACING_ISR
+	sys_trace_isr_exit();
```

Comment:
at least for `sys_trace_thread` and `sys_trace_sem` functions, we have them compile to nothing if tracing is disabled.

can we do the same for ISR, so we don't have to wrap these invocations in ifdefs?

Reply:
yeah, i dont like the too many ifdefs either, this will move the ifdef into the function however, which I think is fine..